ckanext/s3filestore/controller.py

```diff
@@ -101,10 +105,25 @@ def filesystem_resource_download(self, id, resource_id, filename=None):
         if rsc.get('url_type') == 'upload':
             upload = DefaultResourceUpload(rsc)
             try:
-                return upload.download(rsc['id'], filename)
-            except OSError:
+                if hasattr(upload, 'download'):
+                    return upload.download(rsc['id'], filename)
+                else:
+                    filepath = upload.get_path(rsc['id'])
```

Comment:
why add back this logic. 

With how we updated the IUploader interface, the S3 IUploader plugin will defer to the internal file system if it can't be found in s3 so this logic you have added makes it double look and not really helping..

So its not in the controller.

Reply:
See https://github.com/qld-gov-au/ckanext-s3filestore/blob/master/ckanext/s3filestore/uploader.py#L359 that does this logic hand back to core ckan module.

Reply:
This is meant to handle the case where we're running on vanilla CKAN instead of our fork (eg in our tests). Note that the line you refer to uses the `download` function that our fork adds.

Reply:
please add comment then.
Also could you see/update the pr so our 'upgrade' to the api is included in 2.9.